Remotion/Web/UnitTests/Core/Globalization/ResourceManagerExtensionsTest.cs

```diff
@@ -0,0 +1,248 @@
+          .Setup (mock => mock.TryGetString (resourceId, out outValue))
+          .Returns (true)
+          .Verifiable();
+
+      var result = _resourceManagerMock.Object.GetWebStringOrDefault (resourceId, WebStringType.PlainText);
+
+      _resourceManagerMock.Verify();
+      Assert.That (result.HasValue, Is.True);
+      Assert.That (result.Value.GetValue(), Is.EqualTo ("Test"));
+      Assert.That (result.Value.Type, Is.EqualTo (WebStringType.PlainText));
+    }
+
+    [Test]
+    public void GetWebStringOrDefault_WithNonExistingResource_ReturnsNull ()
+    {
+      var result = _resourceManagerMock.Object.GetWebStringOrDefault (c_fakeResourceId, WebStringType.PlainText);
+
+      Assert.That (result.HasValue, Is.False);
+    }
+
+    [Test]
+    public void GetWebString_Enum_WithExistingResourceAndWebStringTypeEncoded_ReturnsWebString ()
+    {
+      var enumValue = EnumWithMultiLingualNameAttribute.ValueWithLocalizedName;
+      var enumResourceID = ResourceIdentifiersAttribute.GetResourceIdentifier (enumValue);
+      var outValue = "Test";
+
+      _resourceManagerMock
+          .Setup (mock => mock.TryGetString (enumResourceID, out outValue))
+          .Returns (true)
+          .Verifiable();
+
+      var result = _resourceManagerMock.Object.GetWebString (enumValue, WebStringType.Encoded);
+
+      _resourceManagerMock.Verify();
+      Assert.That (result.GetValue(), Is.EqualTo ("Test"));
+      Assert.That (result.Type, Is.EqualTo (WebStringType.Encoded));
+    }
+
+    [Test]
+    public void GetWebString_Enum_WithExistingResourceAndWebStringTypePlaintext_ReturnsWebString ()
+    {
+      var enumValue = EnumWithMultiLingualNameAttribute.ValueWithLocalizedName;
+      var enumResourceID = ResourceIdentifiersAttribute.GetResourceIdentifier (enumValue);
+      var outValue = "Test";
+
+      _resourceManagerMock
+          .Setup (mock => mock.TryGetString (enumResourceID, out outValue))
+          .Returns (true)
+          .Verifiable();
+
+      var result = _resourceManagerMock.Object.GetWebString (enumValue, WebStringType.PlainText);
+
+      _resourceManagerMock.Verify();
+      Assert.That (result.GetValue(), Is.EqualTo ("Test"));
+      Assert.That (result.Type, Is.EqualTo (WebStringType.PlainText));
+    }
+
+    [Test]
+    public void GetWebString_Enum_WithNonExistingResourceAndWebStringTypePlaintext_ReturnsWebStringWithResourceIdAsValue ()
+    {
+      var enumValue = EnumWithMultiLingualNameAttribute.ValueWithLocalizedName;
+      var enumResourceID = ResourceIdentifiersAttribute.GetResourceIdentifier (enumValue);
+
+      var result = _resourceManagerMock.Object.GetWebString (enumValue, WebStringType.PlainText);
+
+      Assert.That (result.GetValue(), Is.EqualTo (enumResourceID));
+      Assert.That (result.Type, Is.EqualTo (WebStringType.PlainText));
+    }
+
+    [Test]
+    public void GetWebString_Enum_WithNonExistingResourceAndWebStringTypeEncoded_ReturnsWebStringWithResourceIdAsValue ()
```

Comment:
Naming: id is always uppercase `ID` unless (debatable) the source API already uses the modern spelling.